pkg/apis/core/validation/shoot.go

```diff
@@ -323,7 +320,6 @@ func ValidateShootSpecUpdate(newSpec, oldSpec *core.ShootSpec, newObjectMeta met
 	}
 
 	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Region, oldSpec.Region, fldPath.Child("region"))...)
-	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.CloudProfileName, oldSpec.CloudProfileName, fldPath.Child("cloudProfileName"))...)
```

Comment:
I think we can't simply remove this check. The immutability must still be given, no matter how the reference is defined.

Reply:
As discussed via chat: To make this PR more wholesome I already added an implementation for getting the root CloudProfile and validate that by switching here, the applied CloudProfile after an Update is always either the root or a descendant thereof